src/db.cpp

```diff
@@ -80,6 +80,23 @@ bool DbTable::next(uint32_t* type, string* data, SymmCipher* key)
     return false;
 }
 
+void DbTable::checkTransaction()
+{
+    if (mCheckAlwaysTransacted)
+    {
+        assert(mCurrentTransaction);  // if this fails, we should have started a DBTableTransactionCommitter higher in the call stack
+        if (mCurrentTransaction)
+        {
+            mCurrentTransaction->beginOnce();
+        }
```

Comment:
The assert should be `mCurrentTransaction && mCurrentTransaction.started`, shouldn't it?
Similarly the `if`, should do `mCurrentTransaction && !mCurrentTransaction.started`, although that's already checked within `beginOnce`

Reply:
Usually that would be the case, but on the first operation since the committer was created, `started` would be false, and the `beginOnce()` will start the transaction at this point.   The programmer can choose to start it themselves by calling `beginOnce` manually of course, but sometimes (maybe usually?) it's better to defer that, and let the first database operation start the transaction (for those tables with checkAlwaysTransacted turned on) in case the transaction is not needed at all - there may be some conditionals on the database operation, or a loop that might have 0 iterations, and if we can avoid the begin/commit entirely in those cases, it's much better, for performance and to avoid the begin/commit logging also.     One other point, keeping that logic in the committer object allows those to still be used occasionally for a db table that doesn't have the checkAlwaysTransacted turned on, should that be useful.  cheers!